From e218b420aed0979e6d41572a2bf68d4fa1cd74c3 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:42:24 -0400
Subject: [infiniband] cxgb4: Add missing neigh_release in LE-Workaround path

Message-id: <1410475447-94392-6-git-send-email-svemuri@redhat.com>
Patchwork-id: 93793
O-Subject: [RHEL7.1 PATCH BZ 1124947 005/108] RDMA/cxgb4: Add missing neigh_release in LE-Workaround path
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Signed-off-by: Kumar Sanghvi <kumaras@chelsio.com>
Signed-off-by: Hariprasad Shenai <hariprasad@chelsio.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 0f0132001fd239bb67c1f68436b95cc79de89736)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/cxgb4/cm.c b/drivers/infiniband/hw/cxgb4/cm.c
index 4512687..d286bde 100644
--- a/drivers/infiniband/hw/cxgb4/cm.c
+++ b/drivers/infiniband/hw/cxgb4/cm.c
@@ -3352,6 +3352,7 @@ static int rx_pkt(struct c4iw_dev *dev, struct sk_buff *skb)
   goto free_dst;
  }
 
+ neigh_release(neigh);
  step = dev->rdev.lldi.nrxq / dev->rdev.lldi.nchan;
  rss_qid = dev->rdev.lldi.rxq_ids[pi->port_id * step];
  window = (__force u16) htons((__force u16)tcph->window);
-- 
1.7.1